src/azure/cli/_profile.py

```diff
@@ -334,9 +331,7 @@ def persist_cached_creds(self):
 
             all_creds.extend(self._service_principal_creds)
             cred_file.write(json.dumps(all_creds))
-
-        if new_file:
-            os.chmod(self._token_file, 0o600)
+            cred_file.truncate()#to avoid reopening and write
```

Comment:
doesn't this always remove the contents you just wrote one line above? if you need file to be truncated when you open it, you can use `os.O_TRUNC`.